Core/include/Acts/TrackFitting/detail/GsfActor.hpp

```diff
@@ -525,7 +532,7 @@ struct GsfActor {
       auto& cmp = *res;
       auto freeParams = cmp.pars();
       cmp.jacToGlobal() = surface.boundToFreeJacobian(
-          state.geoContext, freeParams.template segment<3>(eFreePos0),
+          state.stepping.geoContext, freeParams.template segment<3>(eFreePos0),
```

Comment:
This is not strictly required to make this work, right?

Reply:
Overlooked this one. Reverted

Reply:
It seems a bit odd that we use three different spots for the geometry context. I see
- `state.stepping.geoContext`
- `state.options.geoContext`
- `state.geoContext`

Reply:
would it make sense to have a struct containing all the contexts passing by argument to the propagator?

Reply:
I think I had a PR somewhere but people were more in favor of having the context in the options / states